zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/Schema.java

```diff
@@ -126,18 +185,26 @@ static boolean hasUpgrade2_remoteService(KeyspaceMetadata keyspaceMetadata) {
     return keyspaceMetadata.getTable(TABLE_SERVICE_REMOTE_SERVICES).isPresent();
   }
 
-  static void applyCqlFile(String keyspace, CqlSession session, String resource) {
-    Version version = ensureVersion(session.getMetadata());
+  static void applyCqlFile(Version version, String keyspace, CqlSession session, String resource) {
     for (String cmd : resourceToString(resource).split(";", 100)) {
       cmd = cmd.trim().replace(" " + DEFAULT_KEYSPACE, " " + keyspace);
       if (cmd.isEmpty()) continue;
       cmd = reviseCQL(version, cmd);
-      session.execute(cmd);
+      try {
+        session.execute(cmd);
+      } catch (InvalidQueryException e) {
+        // Add context so it is obvious which line was wrong
+        String message = String.format("Failed to execute [%s]: %s", cmd, e.getMessage());
+        // Ensure we can look at logs to see the problem. Otherwise, it may only
+        // be visible in API error responses, such as /health or /api/v2/traces.
+        LOG.error(message);
+        throw new RuntimeException(message, e);
+      }
     }
   }
 
   static String reviseCQL(Version version, String cql) {
-    if (version.getMajor() == 4) {
+    if (version.getMajor() >= 4) {
```

Comment:
PS I won't be making a cassandra 5 image as it is still beta, but FYI with this people can try on own.. provided they enable SASI in cassandra.yaml like this:

```diff
diff --git a/zipkin-storage/cassandra/pom.xml b/zipkin-storage/cassandra/pom.xml
index 417887e2b..c6e67df63 100644
--- a/zipkin-storage/cassandra/pom.xml
+++ b/zipkin-storage/cassandra/pom.xml
@@ -99,5 +99,11 @@
       <artifactId>jnr-posix</artifactId>
       <version>3.1.19</version>
     </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>cassandra</artifactId>
+      <version>${testcontainers.version}</version>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
 </project>
diff --git a/zipkin-storage/cassandra/src/test/java/zipkin2/storage/cassandra/CassandraContainer.java b/zipkin-storage/cassandra/src/test/java/zipkin2/storage/cassandra/CassandraContainer.java
index 536738e49..96cf46257 100644
--- a/zipkin-storage/cassandra/src/test/java/zipkin2/storage/cassandra/CassandraContainer.java
+++ b/zipkin-storage/cassandra/src/test/java/zipkin2/storage/cassandra/CassandraContainer.java
@@ -16,9 +16,8 @@ import java.util.List;
 import java.util.Optional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.output.Slf4jLogConsumer;
-import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
@@ -28,14 +27,16 @@ import static zipkin2.storage.cassandra.ITCassandraStorage.SEARCH_TABLES;
 import static zipkin2.storage.cassandra.Schema.TABLE_DEPENDENCY;
 import static zipkin2.storage.cassandra.Schema.TABLE_SPAN;
 
-class CassandraContainer extends GenericContainer<CassandraContainer> {
+class CassandraContainer
+  extends org.testcontainers.containers.CassandraContainer<CassandraContainer> {
   static final Logger LOGGER = LoggerFactory.getLogger(CassandraContainer.class);
   CqlSession globalSession;
 
   CassandraContainer() {
-    super(parse("ghcr.io/openzipkin/zipkin-cassandra:3.1.0"));
-    addExposedPort(9042);
-    waitStrategy = Wait.forHealthcheck();
+    super(parse("cassandra:5.0-beta1"));
+    // edited to sasi_indexes_enabled: true
+    withCopyFileToContainer(
+      MountableFile.forClasspathResource("cassandra.yaml", 0_600),
+      "/etc/cassandra/cassandra.yaml");
     withLogConsumer(new Slf4jLogConsumer(LOGGER));
   }
```

Reply:
on my laptop all tests are green with above, fyi, with cassandra 5